code/WorkInProgress/blueprints.dm

```diff
@@ -46,184 +68,305 @@
 
 	attackby(obj/item/W, mob/user)
 		if(istype(W, /obj/item/blueprint))
-			if(currentBp)
+			if(src.currentBp)
 				boutput(user, "<span class='alert'>Theres already a blueprint in the machine.</span>")
 				return
 			else
 				boutput(user, "<span class='notice'>You insert the blueprint into the machine.</span>")
 				user.drop_item()
 				W.set_loc(src)
-				currentBp = W
+				src.currentBp = W
 				return
-		else if (istype(W, /obj/item/sheet))
-			boutput(user, "<span class='notice'>You insert the sheet into the machine.</span>")
+		else if (istype(W, /obj/item/sheet) || istype(W, /obj/item/material_piece))
+			boutput(user, "<span class='notice'>You insert the material into the machine.</span>")
 			user.drop_item()
 			W.set_loc(src)
 			return
 		return
 
 	attack_hand(mob/user)
-		if(building)
-			boutput(user, "<span class='alert'>The machine is currently constructing something. Best not touch it until it's done.</span>")
+		if(src.building && !src.Paused)
+			if (tgui_alert(user, "Pause the construction?", "ABCU", list("Yes", "No")) == "Yes")
+				src.pauseBuild()
+				return
 			return
```

Comment:
```suggestion
```
superfluous return

Reply:
This return prevents option list from running, if user declines or closes the pause alert.

Reply:
no it doesn't, look at line 92

Reply:
I don't understand. It's an early return, meant to prevent the option menu from appearing, after the user declines to pause construction.

Reply:
Line 92 is also an early return

Reply:
Okay now I see. It's line 91 that's superfluous, not 92. I'll remove that